### PR TITLE
1242, 1274, ans 1275 - Sorted threads and messages by time and added text subject.

### PR DESCRIPTION
**Pull request by kellykelly3**
I forgot to rename the branch before I pushed so just ignore that branch name please.

### CODE REVIEW

File: Core/src/org/sleuthkit/autopsy/communications/relationships/MessagesChildNodeFactory.java [hunk 1]
```diff
@@ -18,17 +18,17 @@
...
 import org.openide.nodes.ChildFactory;
 import org.openide.nodes.Node;
-import org.sleuthkit.autopsy.casemodule.Case;
-import org.sleuthkit.autopsy.casemodule.NoCurrentCaseException;
 import org.sleuthkit.autopsy.coreutils.Logger;
 import org.sleuthkit.datamodel.BlackboardArtifact;
 import org.sleuthkit.datamodel.BlackboardAttribute;
-import org.sleuthkit.datamodel.CommunicationsManager;
+import static org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME;
+import static org.sleuthkit.datamodel.BlackboardAttribute.ATTRIBUTE_TYPE.TSK_DATETIME_START;
```
Comment (wschaeferB) [inline Core/src/org/sleuthkit/autopsy/communications/relationships/MessagesChildNodeFactory.java]: Unused imports

File: Core/src/org/sleuthkit/autopsy/communications/relationships/ThreadNode.java [hunk 2]
```diff
@@ -18,11 +18,19 @@
  */
 package org.sleuthkit.autopsy.communications.relationships;
 
+import java.util.logging.Level;
 import javax.swing.Action;
 import org.openide.nodes.AbstractNode;
 import org.openide.nodes.Children;
 import org.openide.nodes.Sheet;
+import org.openide.util.Exceptions;
```
Comment (wschaeferB) [inline Core/src/org/sleuthkit/autopsy/communications/relationships/ThreadNode.java]: Unused import

File: Core/src/org/sleuthkit/autopsy/communications/relationships/ThreadNode.java [hunk 3]
```diff
@@ -31,6 +39,10 @@
  */
 final class ThreadNode extends AbstractNode{
     
+    private static final Logger logger = Logger.getLogger(ThreadNode.class.getName());
+    
+    final static int MAX_SUBJECT_LENGTH = 120;
```
Comment (wschaeferB) [inline Core/src/org/sleuthkit/autopsy/communications/relationships/ThreadNode.java]: can be private i think

**Comment by rcordovano** [comment]
@kellykelly3, it looks like the three Codacy issues could be resolved; at least the switch issues should be resolved. Thanks!

**Comment by kellykelly3** [comment]
> @kellykelly3, it looks like the three Codacy issues could be resolved; at least the switch issues should be resolved. Thanks!

I've never used a break in my defaults (which were added to make codacy happy), but I'll add the breaks.

**Pull request merged**